src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -396,21 +445,131 @@ private Properties buildDefaultServerProperties() {
+
+        // Define Callback Handlers for OAUTHBEARER
+        final String serverCallbackHandler = "io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler";
+
+        properties.setProperty("listener.name.plaintext.oauthbearer.sasl.server.callback.handler.class", serverCallbackHandler);
+        properties.setProperty("listener.name.broker1.oauthbearer.sasl.server.callback.handler.class", serverCallbackHandler);
+        properties.setProperty("listener.name.controller.oauthbearer.sasl.server.callback.handler.class", serverCallbackHandler);
+
+        final String clientSideCallbackHandler = "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler";
+
+        // Optionally, define client-side callback handlers if using inter-broker communication
+        properties.setProperty("listener.name.plaintext.oauthbearer.sasl.login.callback.handler.class", clientSideCallbackHandler);
+        properties.setProperty("listener.name.broker1.oauthbearer.sasl.login.callback.handler.class", clientSideCallbackHandler);
+        properties.setProperty("listener.name.controller.oauthbearer.sasl.login.callback.handler.class", clientSideCallbackHandler);
+    }
+
+    /**
+     * Overrides the default Kafka server properties with the provided overrides.
+     * If the overrides map is null or empty, it simply returns the default properties as a string.
+     *
+     * @param defaultProperties The default Kafka server properties.
+     * @param overrides         The properties to override. Can be null.
+     * @return A string representation of the combined server properties.
+     */
     private String overrideProperties(Properties defaultProperties, Map<String, String> overrides) {
-        // Apply overrides
-        overrides.forEach(defaultProperties::setProperty);
+        // Check if overrides are not null and not empty before applying them
+        if (overrides != null && !overrides.isEmpty()) {
+            overrides.forEach(defaultProperties::setProperty);
+        }
 
         // Write properties to string
         StringWriter writer = new StringWriter();
         try {
             defaultProperties.store(writer, null);
         } catch (IOException e) {
-            throw new UncheckedIOException(e);
+            throw new UncheckedIOException("Failed to store Kafka server properties", e);
         }
 
         return writer.toString();
     }
 
+    /**
+     * Sets the 'super.users' property in the Kafka server properties.
+     *
+     * @param properties The Kafka server properties.
+     */
+    private void setSuperUsersIntoProperties(Properties properties) {
+        if (this.superUsers != null && !this.superUsers.isEmpty()) {
+            String superUsersProperty = this.superUsers.stream()
+                .map(user -> "User:" + user)
+                .collect(Collectors.joining(";"));
+            properties.setProperty("super.users", superUsersProperty);
+        }
+    }
```

Comment:
You have the ability to set `super.users` but I don't see in any configuration that you specify the `authorizer.class.name`. Without specifying that in the Kafka broker configuration there is no active authorization in place, and setting `super.users` becomes irrelevant (ignored).